Index: mt76x2e-3.0.4.0/rlt_wifi/include/oid.h
===================================================================
--- mt76x2e-3.0.4.0.orig/rlt_wifi/include/oid.h
+++ mt76x2e-3.0.4.0/rlt_wifi/include/oid.h
@@ -1297,8 +1297,8 @@ typedef struct _RT_802_11_MAC_ENTRY {
 //sync with MT7620
 //#if 1//def RTMP_RBUS_SUPPORT
 #ifndef CUSTOMER_DCC_FEATURE
-	int StreamSnr[3];				/* BF SNR from RXWI. Units=0.25 dB. 22 dB offset removed */
-	int SoundingRespSnr[3];			/* SNR from Sounding Response. Units=0.25 dB. 22 dB offset removed */
+	SHORT StreamSnr[3];				/* BF SNR from RXWI. Units=0.25 dB. 22 dB offset removed */
+	SHORT SoundingRespSnr[3];			/* SNR from Sounding Response. Units=0.25 dB. 22 dB offset removed */
 /*	SHORT TxPER;	*/					/* TX PER over the last second. Percent */
 /*	SHORT reserved;*/
 #endif /* RTMP_RBUS_SUPPORT */
