Index: mt7603e-4.0.1.0/rlt_wifi/common/cmm_info.c
===================================================================
--- mt7603e-4.0.1.0.orig/rlt_wifi/common/cmm_info.c
+++ mt7603e-4.0.1.0/rlt_wifi/common/cmm_info.c
@@ -1362,7 +1362,9 @@ INT	Set_WscGenPinCode_Proc(RTMP_ADAPTER
     PWSC_CTRL   pWscControl = NULL;
     POS_COOKIE  pObj = (POS_COOKIE) pAd->OS_Cookie;
     UCHAR	    apidx = pObj->ioctl_if;
-    
+    struct file *filePtr;
+    char buffer[512] = {0};
+	
 #ifdef CONFIG_AP_SUPPORT
 	IF_DEV_CONFIG_OPMODE_ON_AP(pAd)
 	{
@@ -1404,6 +1406,24 @@ INT	Set_WscGenPinCode_Proc(RTMP_ADAPTER
 
 
 	DBGPRINT(RT_DEBUG_TRACE, ("Set_WscGenPinCode_Proc:: Enrollee PinCode\t\t%08u\n", pWscControl->WscEnrolleePinCode));
+	printk("Set_WscGenPinCode_Proc:: Enrollee PinCode\t\t%08u\n", pWscControl->WscEnrolleePinCode);
+	if(4 == pWscControl->WscEnrolleePinCodeLen )
+	  sprintf(buffer,"%04u\n",pWscControl->WscEnrolleePinCode);
+	else
+	  sprintf(buffer,"%08u\n",pWscControl->WscEnrolleePinCode);
+	
+	filePtr = filp_open("/tmp/.pincode", O_TRUNC|O_CREAT, 0644);
+	if (IS_ERR(filePtr))
+	{
+	  printk("open /tmp/.pincode error\n");
+	} 
+	else
+	{
+	  filePtr->f_op->write(filePtr, buffer, strlen(buffer), &filePtr->f_pos);
+	  filp_close(filePtr, NULL);
+	}
+
+
 
 	return TRUE;
 }
